crates/evm/src/lib.rs

```diff
@@ -121,3 +124,123 @@ pub trait ConfigureEvmEnv: Send + Sync + Unpin + Clone + 'static {
         fill_block_env(block_env, chain_spec, header, after_merge);
     }
 }
+
+/// Trait for managing the EVM context.
+/// Allows to build an EVM Executor with the given context.
+pub trait EvmContext<'a, EXT, DB: Database> {
+    /// The executor produced with the context set.
+    type Executor: EvmExecutor;
+
+    /// Sets the EVM database.
+    #[must_use]
+    fn with_db<DBN: Database + 'a>(self, db: DBN) -> impl EvmContext<'a, EXT, DBN>;
+
+    /// Sets the EVM environment.
+    #[must_use]
+    fn with_env(self, env: Env) -> Self;
+
+    /// Sets the EVM configuration environment.
+    #[must_use]
+    fn with_cfg_env(self, cfg: CfgEnv) -> Self;
+
+    /// Sets the block environment.
+    #[must_use]
+    fn with_block_env(self, block: BlockEnv) -> Self;
+
+    /// Sets the transaction environment.
+    #[must_use]
+    fn with_tx_env(self, tx: TxEnv) -> Self;
+
+    /// Sets the specification (hardfork) for the EVM instance.
+    #[must_use]
+    fn with_spec_id(self, spec_id: SpecId) -> Self;
+
+    /// Sets the inspector for the EVM instance.
+    #[must_use]
+    fn with_inspector<EXTN: 'a>(self, inspector: EXTN) -> impl EvmContext<'a, EXTN, DB>;
+
+    /// Returns the EVM executor.
+    fn executor(self) -> Self::Executor;
+}
+
+/// A context for configuring a Revm EVM instance.
+#[allow(missing_debug_implementations)]
+pub struct RevmContext<'a, EXT, DB: Database> {
+    context: revm::Context<EXT, DB>,
+    handler: revm::Handler<'a, revm::Context<EXT, DB>, EXT, DB>,
+}
```

Comment:
hmm, this looks a bit odd

Reply:
Agreed, I started this out to see if I could implement the `EvmContext` trait for `revm::Evm` but it proved impossible. I think the best path is to remove it